src/main/java/sirius/biz/translations/BasicTranslations.java

```diff
@@ -0,0 +1,174 @@
+        }
+
+        Optional<T> translation = fetchTranslation(field, lang);
+        return translation.map(t -> t.getTranslationData().getText());
+    }
+
+    /**
+     * Returns the translated text for the given field and language. Uses fallback language if no translation is found.
+     * If the fallback also has no translation, the default text from the owner entity is returned.
+     *
+     * @param field        {@link Mapping} of the translated field
+     * @param lang         language code
+     * @param fallbackLang language code for fallback
+     * @return translated text for the given language, fallback, or the default text from owner entity
+     */
+    public String getRequiredText(Mapping field, String lang, String fallbackLang) {
+        Optional<String> text = getText(field, lang);
+        if (!text.isPresent()) {
+            text = getText(field, fallbackLang);
+            if (!text.isPresent()) {
+                return fetchDefaultText(field);
+            }
+        }
+        return text.get();
+    }
+
+    /**
+     * Fetches the default text by querying {@link sirius.db.mixing.FieldLookupCache} for the owner entity's given field.
+     *
+     * @param field {@link Mapping field} to fetch the the text from
+     * @return the text as String, or null if lookup was unsuccessful
+     */
+    protected String fetchDefaultText(Mapping field) {
+        return fieldLookupCache.lookup(owner.getClass(), owner.getId(), field).asString();
+    }
+
+    /**
+     * Returns all available translated texts for the given field.
+     *
+     * @param field {@link Mapping} of the translated field
+     * @return a {@link Map} of all translated texts with corresponding language code as key, empty map if none present
+     */
+    public Map<String, String> getAllTexts(Mapping field) {
+        Optional<List<T>> translations = fetchAllTranslations(field);
+        if (!translations.isPresent()) {
+            return new HashMap<>();
+        }
+        Map<String, String> translatedTexts = new HashMap<>();
+        for (T t : translations.get()) {
+            translatedTexts.put(t.getTranslationData().getLang(), t.getTranslationData().getText());
+        }
+        return translatedTexts;
+    }
+
+    /**
+     * Looks up the translation entity for the database in use (e.g. MongoDB or SQL).
+     *
+     * @param field {@link Mapping} of the translated field
+     * @param lang  language code
+     * @return the resolved translation entity wrapped as optional or an empty optional if the value couldn't be resolved
+     */
+    protected abstract Optional<T> fetchTranslation(Mapping field, String lang);
+
+    /**
+     * Looks up and returns all available {@link Translation translation} entities for the given field
+     *
+     * @param field {@link Mapping} of the translated field
+     * @return {@link Optional} with a {@link java.util.List} of all {@link Translation translations}, or empty if none are found
```

Comment:
Why not empty list instead of empty optional? Saves a lot of extra code in the calls to this method

Reply:
Thanks for the hint! The optional was definitely unnecessary here